modules/python/src2/gen2.py

```diff
@@ -422,6 +423,10 @@ def __init__(self, classname, name, decl, isconstructor, isphantom=False):
         self.array_counters = {}
         for a in decl[3]:
             ainfo = ArgInfo(a)
+            # Do not treat blob as an image while passing
+            # from python to cpp if passed from setInput()
+            if name=='setInput' and a[1]=='blob':
+                ainfo.isImage=False
```

Comment:
In general, this approach is a breaking change of existed API.

@l-bat Could you try to apply workaround similar to #20480? Something like:

```.python
input = np.expand_dims(input, axis=2)  # Workaround for: https://github.com/opencv/opencv/issues/19091
```

with comment with reference on the issue #19091.